easyeffects: 6.1.3 -> 6.2.3

MarekPasnikowski opened the pull request:
1. Updated to the newest version.
2. Added some build and runtime dependencies according to build errors.

###### Motivation for this change
Fixes: #158476
Fixes: https://github.com/NixOS/nixpkgs/issues/155588

https://github.com/wwmm/easyeffects/blob/v6.2.3/CHANGELOG.md

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Commit:
easyeffects: 6.1.3 -> 6.2.3

1. Updated to the newest version.
2. Added some build and runtime dependencies according to build errors.

jtojnar commented:
The `substituteInPlace` can be removed.

jtojnar commented on pkgs/applications/audio/easyeffects/default.nix:
```diff
@@ -49,15 +52,18 @@ stdenv.mkDerivation rec {
     ninja
     pkg-config
     python3
+    tbb
```
How did you determine it goes to `nativeBuildInputs`?

Obikawa replied:
Good catch - when I made this change, I was under impression that native inputs meant "build time" dependencies. But after thinking about it in the cross-compilation context (which I had learned later) I am not sure what should go where anymore. Is there an algorithm to accurately decide this?

jtojnar replied:
That is about right `nativeBuildInputs` is about dependencies used, or rather, executed at build time. So mostly programs used for locating dependencies, build tools, linkers, documentation generators… `tbb` is a library and while it is possible to build and run an executable linking a library at build time, it is quite rare. So unless you have a reason to suspect a library to be used this way (e.g. `libxslt` contains `xsltproc` program often used for building man pages from DocBook files), it is most likely a runtime dependency belonging to `buildInputs`.

See also https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md

Obikawa replied:
Is it safe to always put any new dependency in buildInputs first? And only when it breaks move it to nativeBuildInputs?

Obikawa replied:
> so it should not be left in the code

Does `it` reference to the `strictDeps` call? I am roughly understanding what you wrote as "Use `strictDeps` in testing, but make sure it is removed in the final code."

jtojnar commented on pkgs/applications/audio/easyeffects/default.nix:
```diff
@@ -49,15 +52,18 @@ stdenv.mkDerivation rec {
     fftwFloat
+    fmt
     glib
     glibmm
     gtk4
     gtkmm4
```
From [changelog](https://github.com/wwmm/easyeffects/blob/v6.2.3/CHANGELOG.md):

> GTKMM and GLIBMM are not a dependency anymore. We now use gtk4 directly.

Commit:
remove unneeded dependencies

1. GTKMM
2. GLIBMM

Commit:
attempt to fix removed glibmm

Removal of glibmm caused an unexpected call. Hopefully, hiding the substituteInPlace (which I was told was no longer needed) fixes it.

Commit:
include glibmm

Apparently, something still requires glibmm. Put it back in the inputs.

jtojnar commented on pkgs/applications/audio/easyeffects/default.nix:
```diff
@@ -4,11 +4,12 @@
 , fftwFloat
+, fmt
 , glib
 , glibmm
```
This one too.

jtojnar replied:
> anonymous function at /home/jtojnar/Projects/nixpkgs/pkgs/applications/audio/easyeffects/default.nix:1:1 called with unexpected argument 'glibmm'

means that you also need to remove `glibmm` from being passed in `all-packages.nix`.

Commit:
attempt to edit the all-packages.nix file

Removed the call to glibmm in easyeffects.

Commit:
cleanup glibmm

Removed GLIBMM from all-packages' call to EasyEffects.

Commit:
transfer tbb to buildInputs

jtojnar commented on pkgs/applications/audio/easyeffects/default.nix:
```diff
@@ -69,14 +70,15 @@ stdenv.mkDerivation rec {
     chmod +x meson_post_install.py
     patchShebangs meson_post_install.py
     # https://github.com/wwmm/easyeffects/pull/1205
-    substituteInPlace meson_post_install.py --replace "gtk-update-icon-cache" "gtk4-update-icon-cache"
+    # substituteInPlace meson_post_install.py --replace "gtk-update-icon-cache" "gtk4-update-icon-cache"
```
You can just drop this. This change was merged in 6.1.4.

Obikawa replied:
Dropped.

Commit:
remove substituteInPlace call

jtojnar commented:
Now it fails on missing

```
src/meson.build:122:0: ERROR: Dependency "sigc++-3.0" not found, tried pkgconfig
```

jtojnar commented:
Could you please also squash everything into `easyeffects: 6.1.3 -> 6.2.3`?

Obikawa commented:
Understood. I have to learn how to squash in GitHub and add libsigcxx30.

I need to do something time-sensitive in fifteen minutes, so expect a longer response time while I get everything in order.

Commit:
include libsigcxx

Commit:
fix libsigcxx version to libsigcxx30

Obikawa commented:
With regards to squashing:

After reading some documentation and internet articles, I got the impression that this option is available only to those with write access to a given repository. I do not have this capability on this repository.

Do you not have the `squash and merge` option available?

jtojnar commented:
You only need to squash in your branch in your repository (https://github.com/Obikawa/nixpkgs/tree/easyeffects-update). You can use `git rebase --interactive master` to rebase onto `master` branch (or whatever other branch you want).

In this case, using “Rebase and squash” will work since we want to produce only a single logical commit, it is useless for anything else. The disadvantage is that you will not be able to use `git branch --merged` to find and delete merged branches in your local repository since “rebase and squash” will not update your own repository. But if you do not use that workflow, I guess that is fine.

Obikawa commented:
It is fine:

1. The branch is only for the purpose of getting the update merged with upstream and will be thrown away after that.
2. I have had trouble using my hardware key (Trezor) to authenticate with GitHub, so I settled with web-only workflow. The GitHub's web interface does not seem to have any option to squash a branch.

jtojnar commented:
Thanks.

jtojnar commented:
Since pipewire is backported we need to backport this as well.
